### PR TITLE
feat: add rudimentry --list and --info flags + minor cleanup

### PR DESCRIPTION
**Pull request by coolaj86**
Re: #698, #708 

Also:
- [x] lint & fmt
- [x] unexpand $HOME in printed output
- [x] guard xattr 

## --list

```sh
webi --list
```

```text
[warn] the format of --list output may change

atomicparsley
awless
archiver
# ...
watchexec
zoxide
yq
```

## --info

```sh
webi --info ssh-pubkey
webi --info node
```

```text
'ssh-pubkey' is a special case that does not have releases
```

```json5
// Stable 'node' releases:
[
  {
    "name": "node-v20.9.0-aix-ppc64.tar.gz",
    "version": "20.9.0",
    "lts": true,
    "channel": "stable",
    "date": "2023-10-24",
    "os": "aix",
    "arch": "ppc64",
    "ext": "tar.gz",
    "download": "https://nodejs.org/download/release/v20.9.0/node-v20.9.0-aix-ppc64.tar.gz"
  }
]
```

### CODE REVIEW

**Review by jojobyte — APPROVED** [review]

File: _webi/template.sh [hunk 1]
```diff
@@ -448,11 +445,10 @@ __bootstrap_webi() {
...
     }
 
-    # move commands from the extracted archive directory to $HOME/.local/opt or $HOME/.local/bin
-    # shellcheck disable=2120
-    # webi_install may be sourced and used elsewhere
+    # move commands from the extracted archive directory
+    # to $HOME/.local/opt or $HOME/.local/bin
     webi_install() {
-        if [ -n "$WEBI_SINGLE" ] || [ "single" = "${1-}" ]; then
+        if test -n "${WEBI_SINGLE}"; then
```
Comment (coolaj86) [inline _webi/template.sh:477]: verified that `webi_install` isn't used externally and removed shellcheck comment

File: _webi/template.sh [hunk 2]
```diff
@@ -472,28 +468,8 @@ __bootstrap_webi() {
             xattr -r -d com.apple.quarantine "$pkg_src" || true
             return 0
         fi
-        # TODO need to test that the above actually worked
```
Comment (coolaj86) [inline _webi/template.sh:475]: It's been a few years. This has never popped up.

File: _webi/bootstrap.sh [hunk 3]
```diff
@@ -42,8 +42,11 @@ set -u
 
 __webi_main() {
 
-    export WEBI_TIMESTAMP="\$(date +%F_%H-%M-%S)"
-    export _webi_tmp="\${_webi_tmp:-\$(mktemp -d -t webi-\$WEBI_TIMESTAMP.XXXXXXXX)}"
+    my_date="\$(date +%F_%H-%M-%S)"
+    export WEBI_TIMESTAMP="\${my_date}"
+    export _webi_tmp="\${_webi_tmp:-\$(
+        mktemp -d -t "webi-\$WEBI_TIMESTAMP.XXXXXXXX"
```
Comment (coolaj86) [inline _webi/bootstrap.sh:48]: It seems that just putting this on its own line got rid of a linter warning.

That may mean that it's a false negative and the command should still be lifted later.

File: _webi/template.sh [hunk 4]
```diff
@@ -598,7 +604,9 @@ __bootstrap_webi() {
 
         webi_link
 
-        _webi_enable_exec
+        if test -z "${pkg_no_exec}"; then
```
Comment (coolaj86) [inline _webi/template.sh:607]: this allows us to skip `xattr` no git cloned directories, which aren't handled properly by `xattr`.

**Review by fidelmeramon-maker — CHANGES_REQUESTED** [review]
- [ ] @

**Pull request merged**